src/components/ContractEventCell.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+  <div class="contract-event-cell">
+    <Suspense>
+      <component
+        :is="contractEventCellComponent"
+        :event="event"
+        :contract-details="contractDetails"/>
+      <template #fallback>
+        Loading...
+      </template>
+    </Suspense>
+  </div>
+</template>
+
+<script setup>
+import { defineAsyncComponent } from 'vue'
+
+const props = defineProps({
+  contractDetails: {
+    type: Object,
+    required: true,
+  },
+  event: {
+    type: Object,
+    required: true,
+  },
+})
+
+const contractEventCellComponent = computed(() =>
+  defineAsyncComponent(async() => {
+    try {
+      return await import(`@/components/ContractEventCell${props.event.eventName}.vue`)
+    } catch {
+      console.error(`Uknown event ${props.event.eventName} in contract ${props.contractDetails.id}`)
+      return defineComponent(() => () => h('span', 'N/A'))
```

Comment:
Since computed property should not modify another variable when getting read, introducing hasError here would be against good practices of vue clean code. We would either need to catch this error in the parent components (where ContractEventCell is used) or define a new vue component that in the template simply shows <span>N/A</span>. I considered all of these options and thought that the currently used approach makes the most sense and is simplistic.

Reply:
Thanks for an explanation, I think it makes sense. Could you point me to that good practices of vue clean code?